add example of using Tables.dictcolumntable (#3387)
JuliaData/DataFrames.jl: docs/src/man/basics.md

   2 │     2      0
```

Sometimes your source data might have a heterogeneous set of columns for each observation.
Here is an example:

```
julia> source = [(type="circle", radius=10), (type="square", side=20)]
2-element Vector{NamedTuple{names, Tuple{String, Int64}} where names}:
 (type = "circle", radius = 10)
 (type = "square", side = 20)
```

If you want to create a data frame from such data containing all columns present in at least

> one of the source observations, and holding `missing` entry if some column is not present then

Reviewer: ```suggestion
one of the source observations, with a `missing` entry if some column is not present then
```

you can use `Tables.dictcolumntable` function to help you create the desired data frame:

```
julia> DataFrame(Tables.dictcolumntable(source))
2×3 DataFrame
 Row │ type    radius   side
     │ String  Int64?   Int64?
─────┼──────────────────────────
   1 │ circle       10  missing
   2 │ square  missing       20
```

> The `Tables.dictcolumntable` role is to make sure that the `DataFrame` constructor gets information

Reviewer 2: ```suggestion
The role of `Tables.dictcolumntable` is to make sure that the `DataFrame` constructor gets information
```
